internal/wal_segment_runner.go

```diff
@@ -47,15 +47,21 @@ type WalSegmentRunner struct {
 	currentWalSegment WalSegmentDescription
 	walFolderSegments map[WalSegmentDescription]bool
 	stopSegmentNo     WalSegmentNo
+	timelineSwitchMap map[WalSegmentNo]*TimelineHistoryRecord
 }
 
 func NewWalSegmentRunner(
 	startWalSegment WalSegmentDescription,
 	segments map[WalSegmentDescription]bool,
 	stopSegmentNo WalSegmentNo,
+	timelineSwitchMap map[WalSegmentNo]*TimelineHistoryRecord,
 ) *WalSegmentRunner {
-	return &WalSegmentRunner{startWalSegment,
-		segments, stopSegmentNo}
+	return &WalSegmentRunner{
+		currentWalSegment: startWalSegment,
```

Comment:
->currentWalSegment: startWalSegment
rename one

Reply:
currentWalSegment is tend to change over iterations, startWalSegment is just the initial value